documents/int256.go

```diff
@@ -0,0 +1,108 @@
+package documents
+
+import (
+	"math/big"
+	"regexp"
+
+	"github.com/centrifuge/go-centrifuge/errors"
+	"github.com/centrifuge/go-centrifuge/utils/byteutils"
+	"github.com/ethereum/go-ethereum/common/math"
+)
+
+// signByteIdx is the index of byte that contains the sign bit of the int256 when represented as 32 bytes
+const signByteIdx = 0
+
+// validInt regex for only allowing integers
+var validInt = regexp.MustCompile(`^[-+]?\d+$`)
+
+// Int256 represents a signed 256 bit integer
+type Int256 struct {
+	v big.Int
+}
+
+// NewInt256 creates a new Int256 given a string
+func NewInt256(n string) (*Int256, error) {
+	if !validInt.MatchString(n) {
+		return nil, errors.NewTypedError(ErrInvalidInt256, errors.New("probably a decimal value: %s", n))
+	}
+
+	nn := new(big.Int)
+	nn, ok := nn.SetString(n, 10)
+	if !ok {
+		return nil, errors.NewTypedError(ErrInvalidInt256, errors.New("probably an arbitrary string: %s", n))
+	}
+
+	if !isValidInt256(*nn) {
+		return nil, errors.NewTypedError(ErrInvalidInt256, errors.New("value: %s", n))
+	}
+	return &Int256{*nn}, nil
+}
+
+// FromBytes converts the a big endian byte slice to an Int256
+func FromBytes(b []byte) (*Int256, error) {
+	if len(b) != 32 {
```

Comment:
Do we have to make it byte 32 all the time? it too much space IMO

Reply:
yes, as the format needs to be bigendian with sign at LSB

Reply:
also it follows ethereum format